src/main/java/build/buildfarm/instance/shard/Util.java

```diff
@@ -183,6 +194,7 @@ static void checkMissingBlobOnInstance(
           @Override
           public void onSuccess(Iterable<Digest> missingDigests) {
             boolean found = Iterables.isEmpty(missingDigests);
+            casHitCounter.inc();
```

Comment:
I'm not sure if this is accurate as this just reports success and can be found or not found (see log message below).

Reply:
I agree with Yuriy. The success here means result list is returned, but could be empty, which means cas misses. Similarly, the Failure means the result list is not returned due to failures, like timeout, which does not necessary means and hit miss.

Reply:
okay, good point.  we're trying to track how often CAS information is unavailable because it indicates more work needing done.  

For example:
 - the re-uploading of CAS data
 - trying to re-fetch CAS from a different worker
 - workerMap needing re-indexed
 - the re-execution of an action (even though it was in the ActionCache it may need re-executed if the outputs were missing)

For those situations, we tried to identify where hits/misses would occur: 
![Blank diagram](https://user-images.githubusercontent.com/1312081/132731926-3b9ae60d-430c-43bc-b81c-ea109b47f0ab.png) 

This was our attempt at tracking (3).  We need to see if there is a better place to hook into

Reply:
Maybe fetchBlobFromWorker in ShardInstance would be a better place to look?

Reply:
Thanks for the comment. Made the changes, can you take a look again?